src/frontend/screens/Settings/MapSettings/BackgroundMapInfo.tsx

```diff
@@ -0,0 +1,330 @@
+          const tile = tilePromise.value;
+          if (!tile.maxzoom) return highest;
+          if (tile.maxzoom > highest) return tile.maxzoom;
+          return highest;
+        }, -1);
+
+        if (maxZoom === -1) throw new Error("no max zoom available");
+
+        setZoom(maxZoom);
+      })
+      .catch(err => {
+        console.log(err);
+        setZoom(null);
+      });
+  }, [id]);
+
+  return (
+    <React.Fragment>
+      <View style={{ flex: 1, backgroundColor: WHITE, paddingBottom: 20 }}>
+        <MapboxGL.MapView
+          styleURL={styleUrl}
+          compassEnabled={false}
+          zoomEnabled={false}
+          logoEnabled={false}
+          scrollEnabled={false}
+          style={{ height: "55%" }}
+        >
+          <MapboxGL.Camera
+            zoomLevel={0}
+            animationDuration={0}
+            animationMode={"linearTo"}
+            allowUpdates={true}
+          />
+        </MapboxGL.MapView>
+        <ScrollView style={styles.container}>
+          {bytesStored && (
+            <Text style={{ color: MEDIUM_GREY }}>
+              {`${convertBytesToMb(bytesStored).toFixed(0)} ${t(m.mb)}`}
+            </Text>
+          )}
+          {zoom === "loading" ? (
+            <Loading />
+          ) : zoom ? (
+            <React.Fragment>
+              <Text>{`${t(m.zoomLevel)}: ${zoom}`}</Text>
+              <Text>{`${t(m.description)}: ${
+                !levelOfDetail ? "" : t(levelOfDetail)
+              }`}</Text>
+            </React.Fragment>
+          ) : null}
+
+          {id !== DEFAULT_MAP_ID && (
+            <Button
+              style={styles.button}
+              fullWidth
+              variant="outlined"
+              onPress={() => sheetRef.current?.snapTo(1)}
+            >
+              <View style={styles.deleteButtonContainer}>
+                <DeleteIcon color={MAPEO_BLUE} />
+                <Text style={styles.deleteButton}>{t(m.deleteMap)}</Text>
+              </View>
+            </Button>
+          )}
+          <Button
+            style={[styles.button, { marginBottom: 20 }]}
+            fullWidth
+            onPress={setStyleAndNavigateHome}
+          >
+            {t(m.useMap)}
+          </Button>
+        </ScrollView>
+      </View>
+
+      <DeleteMapBottomSheet
+        ref={sheetRef}
+        closeSheet={() => {
+          sheetRef.current?.close();
+        }}
+        mapName={name}
+        mapId={id}
+      />
+    </React.Fragment>
+  );
+};
+
+const styles = StyleSheet.create({
+  container: {
+    padding: 20,
+  },
+  button: {
+    marginTop: 20,
+  },
+  deleteButton: {
+    color: MAPEO_BLUE,
+    fontWeight: "700",
+    letterSpacing: 0.5,
+    fontSize: 16,
+  },
+  deleteButtonContainer: {
+    display: "flex",
+    flexDirection: "row",
+    justifyContent: "center",
+    alignItems: "center",
+  },
+});
+
+function allSettled<T>(promises: (Promise<T> | undefined)[]) {
```

Comment:
don't think you need the `undefined` typing and associated code with it unless you know you're dealing with actual `undefined` values in the array. the standard `Promise.allSettled` only accepts an array of promises and breaks otherwise